slice-core/src/main/java/com/cognifide/slice/core/internal/provider/SliceBindingsProvider.java

```diff
@@ -56,4 +57,17 @@ public void addBindings(Bindings bindings) {
 		}
 	}
 
+	private Class<?> getModel(Resource resource) throws ClassNotFoundException {
+		Class<?> modelClass = modelClassNameResolver.getModelClass(resource.getResourceType());
+		while (modelClass == null) {
+			String resourceSuperType = ResourceUtil.findResourceSuperType(resource);
+			if (resourceSuperType == null) {
+				break;
+			}
+			modelClass = modelClassNameResolver.getModelClass(resourceSuperType);
```

Comment:
We must not call this method. It's deprecated. The whole concept needs to be revised here.